Packs/Druva/Integrations/DruvaEventCollector/DruvaEventCollector.py

```diff
@@ -0,0 +1,289 @@
+        set_integration_context(
+            {
+                "Token": raw_token,
+                "expiration_time": (
+                    now + timedelta(seconds=(expires_in_seconds - 60))
+                ).strftime(DATE_FORMAT_FOR_TOKEN),
+            }
+        )
+
+    def _refresh_access_token(self) -> tuple[str, int]:
+        """
+        Since the validity of the Access Token is 30 minutes, this method refreshes it and returns the new token json.
+        returns:
+            - the token
+            - the expiration in seconds
+        """
+        encoded_credentials = base64.b64encode(self.credentials.encode())
+        decoded_credentials = encoded_credentials.decode("utf-8")
+
+        headers = {
+            "Content-Type": "application/x-www-form-urlencoded",
+            "Authorization": f"Basic {decoded_credentials}",
+        }
+        data = {"grant_type": "client_credentials", "scope": "read"}
+
+        try:
+            response_json = self._http_request(
+                method="POST", url_suffix="/token", headers=headers, data=data
+            )
+        except Exception as e:
+            # 400 - "invalid_grant" - reason: invalid Server URL, Client ID or Secret Key.
+            if "invalid_grant" in str(e):
+                raise DemistoException(
+                    "Error in test-module: Make sure Server URL, Client ID and Secret Key are correctly entered."
+                ) from e
+            raise
+        return response_json["access_token"], response_json["expires_in"]
+
+    def search_events(self, tracker: Optional[str] = None) -> dict:
+        """
+        Searches for Druva events.
+
+        Args:
+            tracker: pointer to the last event we got last time
+
+        Returns:
+            dict: List of events
+        """
+        demisto.debug(f'DEC: This is the tracker before encoding: {tracker=}')
+
+        if tracker:
+            encoding_tracker = quote(tracker, safe="!~*'()")  # remove invalid characters from tracker
+            demisto.debug(f'DEC: This is the tracker after encoding: {encoding_tracker=}')
+        else:
+            encoding_tracker = ""
+
+        url_suffix_tracker = f"?tracker={encoding_tracker}" if encoding_tracker else ""
+
+        headers = (self._headers or {}) | {
+            "accept": "application/json"
+        }  # self._headers won't really be None, just for mypy
+        try:
+            response = self._http_request(
+                method="GET",
+                url_suffix=f"/insync/eventmanagement/v2/events{url_suffix_tracker}",
+                headers=headers,
+            )
+        except Exception as e:
+            # 403 - "User is not authorized to access this resource with an explicit deny" - reason: tracker is expired
+            # 400 - "Invalid tracker"
+            raise DemistoException(f"Error in search-events: {str(e)}") from e
+        return response
+
+    def _set_headers(self, token: str):
```

Comment:
add a docstring saying when this is used

Reply:
Done